core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -108,21 +104,24 @@
 
   private synchronized void appendCurrentBuffer(byte[] buf, int offset, int length)
       throws IOException {
-    if (mAvailableBytes < length) {
-      long bytesRequested = mTachyonFS.requestSpace(mBlockId, length - mAvailableBytes);
-      if (bytesRequested + mAvailableBytes >= length) {
-        mAvailableBytes += bytesRequested;
-      } else {
-        mCanWrite = false;
-        throw new IOException(String.format("No enough space on local worker: fileId(%d)"
-            + " blockId(%d) requestSize(%d)", mFile.mFileId, mBlockId, length - mAvailableBytes));
+    initializeLocalFile();
```

Comment:
Oh I see what you mean, I will add the check, since it doesn't make sense if close is called by some edge case and the stream was not writing.